### DISCUSSION TITLE
Problem with browser interactions

### DISCUSSION THREAD

**Discussion by GEJ1** [jspsych/jsPsych#1019] [Q&A] [answered]
Hello everyone!

I have an issue with browser interactions in my experiment.  This is my code:

```js
jsPsych.init({
        
        // timeline: welcome_block,
        timeline: [fullScreen,  chin, instruction,trial, greetings],
        
        on_interaction_data_update: function(data) {
            // get the main trial data
                var trial = jsPsych.currentTrial();
            trial.data.screen_focus = data.event;
        }
```

The error says:

My_experiment.html:290 Uncaught TypeError: Cannot set property 'screen_focus' of undefined
    at Object.on_interaction_data_update (My_experiment.html:290)
    at HTMLDocument.fullscreenchange (jspsych.js:1521)

I have the data without problems but I wondering why this is happening.

Thank you in advance!!

Gustavo.

**Comment by jodeleeuw** [comment, marked as answer]
Hi Gustavo,

I didn't design the trial objects to be modifiable mid-trial like this. Honestly I have no idea what will happen if you try to modify the trial object returned by `jsPsych.currentTrial()`. 

Linking the interaction data with the main timeline data is on our todo list, #495.

In the meantime, I think that you could do something like save the most recent `data.event` in a global variable inside the `on_interaction_data_update` function, and then add a function to the `on_data_update` event that merges an event (if one exists) into the data for that trial. The `on_data_update` function should probably clear the global variable once it has added the data to the trial.

**Reply by GEJ1** [reply]
Thank you for your answer @jodeleeuw !

I will try what you say. I didn'n have experience doing this, but I relied on this tutorial: https://www.youtube.com/watch?v=g7JsikIx0Ho&list=PLtdKTIOUlb42qG962wz30fzlUMibJCGQW&index=11&t
